python/ray/train/_internal/storage.py

```diff
@@ -620,6 +618,24 @@ def get_experiment_dir_name(run_obj: Union[str, Callable, Type]) -> str:
             dir_name = "{}_{}".format(run_identifier, date_str())
         return dir_name
 
+    @staticmethod
+    def _make_checkpoint_dir_name(index: int):
+        """Get the name of the checkpoint directory, given an index."""
+        return f"checkpoint_{index:06d}"
+
+    @staticmethod
+    def _parse_checkpoint_index(checkpoint_dir_name: str):
+        """Parse the index from a checkpoint directory name
+        created by _make_checkpoint_dir_name.
+
+        >>> StorageContext._parse_checkpoint_index("checkpoint_000001")
+        1
+        >>> StorageContext._parse_checkpoint_index("checkpoint_11111111")
+        11111111
+        """
+        assert checkpoint_dir_name.startswith("checkpoint_")
```

Comment:
return None on parsing error?

Reply:
I'm skeptical we need to do parsing in any case... why not set the current trial index from the trial state? It's already in memory, no need to parse it.

Reply:
The driver's trial holds onto a storage context, but it doesn't get the updated checkpoint index at the moment. We could update the driver's copy of the current checkpoint index every time a checkpoint is processed. Then, we'd be tracking this thing separately in 3 places.

```
Driver (trial.storage.current_checkpoint_index) <-- Trainable --> Worker
```

Currently, only the trainable and workers keep the `current_checkpoint_id` synchronized. WDYT?